### DISCUSSION TITLE
Does Inav have capabilities to control a drone programmatically? Like flying by points on a map or controlling a drone from a computer or onboard raspberry pi?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [iNavFlight/inav#8501] [Q&A]
Does Inav have capabilities to control a drone programmatically? Like flying by points on a map or controlling a drone from a computer or onboard raspberry pi?

Can you please share some working examples on how to control a drone programmatically?
Or isn't a inav's speciality and it is better to look to ArduPilot or something else?

**Comment by shudecek** [comment]
I'm interested in something similar. I know i can control an iNav drone with stick inputs and give it a pre-programmed mission.  But i can have an onboard controller like an RPI or Arduino tell it where to go?

**Comment by sensei-hacker** [comment]
For first, the term you want to look for in the documentation is "waypoint mission".

For using a flight computer, look at MSP.
This repo and others by the same person have example code:

https://github.com/stronnag/msp_set_rx

**Comment by stronnag** [comment]
Everything you ask is described in detail on the INAV wiki, for example:

https://github.com/iNavFlight/inav/wiki/INAV-Remote-Management,-Control-and-Telemetry

https://github.com/iNavFlight/inav/wiki/Navigation-modes

And other wiki pages.
